### PR TITLE
Set the dependencies correctly if the previous container steps have b…

### PR DESCRIPTION
**Pull request by mahkoCosmo**
…een stacked

### CODE REVIEW

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt [hunk 1]
```diff
@@ -1624,25 +1624,48 @@ internal fun getCommonEnvVars(
   return (minimalEnvVars + commonEnvVars).toMutableMap()
 }
 
+private fun getNewDependenciesIfExist(
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt:1627]: Method name is not explicit, could you please change it ?

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt [hunk 2]
```diff
@@ -1624,25 +1624,48 @@ internal fun getCommonEnvVars(
...
+  return tmp
+}
+
 private fun stackSolutionContainers(
     containers: MutableList<ScenarioRunContainer>
 ): MutableList<ScenarioRunContainer> {
   val stackedContainers: MutableList<ScenarioRunContainer> = mutableListOf()
   var stackedContainer: ScenarioRunContainer? = null
   var stackedIndex = 1
+  val mergedContainerNames: MutableMap<String, String> = mutableMapOf()
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt:1650]: Can be rewrite `val mergedContainerNames = mutableMapOf<String, String>()` (same remark for line 1647)

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt [hunk 3]
```diff
@@ -1624,25 +1624,48 @@ internal fun getCommonEnvVars(
   return (minimalEnvVars + commonEnvVars).toMutableMap()
 }
 
+private fun getNewDependenciesIfExist(
+    mergedContainerNames: MutableMap<String, String>,
+    container: ScenarioRunContainer
+): ScenarioRunContainer {
+  var tmp: ScenarioRunContainer = container
+  var dependencies: MutableList<String> = mutableListOf()
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt:1632]: Can be rewrite as `val dependencies = mutableListOf<String>()`
 Plus, what append when you have same value several times? 
 How Argo deals with it? Could it be interesting to use `mutableSetOf<String>()` instead ?

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt [hunk 4]
```diff
@@ -1624,25 +1624,48 @@ internal fun getCommonEnvVars(
...
 private fun stackSolutionContainers(
     containers: MutableList<ScenarioRunContainer>
 ): MutableList<ScenarioRunContainer> {
   val stackedContainers: MutableList<ScenarioRunContainer> = mutableListOf()
   var stackedContainer: ScenarioRunContainer? = null
   var stackedIndex = 1
+  val mergedContainerNames: MutableMap<String, String> = mutableMapOf()
   for (container in containers) {
     if (container.solutionContainer != true) {
+      var tmpContainer: ScenarioRunContainer = container
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt:1653]: Could you please change `tmpContainer` name for a more explicit name ?

**Review by jreynard-code — CHANGES_REQUESTED** [review]
Some remarks ;)

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt (hunk 3 above)
Comment (mahkoCosmo) [reply scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt:1632]: dependencies as list is a legacy code, ref ScenarioRunContainer.kt
In Argo's point of view, it expects a list(not a set) so it doesn't care of duplicated name i suppose

**Review by jreynard-code — APPROVED** [review]
LGTM

**Pull request closed without merging**